### DISCUSSION TITLE
Struggles with defining and configuring synchronous client for shovels

### DISCUSSION THREAD

**Discussion by petarzafirovski** [rabbitmq/hop#342] [Q&A]

**Comment by acogoluegnes** [comment]
Could you please provide some steps to reproduce the problem (code and broker configuration if needed)? You can use Docker to start up the broker and please provide the code in block codes not images (use `  ```java ... ``` ` syntax).

**Reply by petarzafirovski** [reply]
@acogoluegnes updated as you said. Also, the exchange (Direct) and Queues , are created via the management UI for now. 
Used the following docker image:
docker run --name rabbitmq-shovel --rm -d -it --hostname my-rabbit -p 15672:15672 -p 5672:5672 bdg91/rabbitmq-shovel:latest

**Comment by lukebakken** [comment]
One option - after declaring the shovel, you could poll the API to see when the queues are empty. At that point, you can delete them.

**Reply by acogoluegnes** [reply]
Polling for what you expect (the shovel creation and the empty queue) is the most appropriate solution indeed. You can poll every second to avoid overflowing the HTTP API and use an appropriate timeout.

Please provide a standalone project / snippet if you want us to try out what you are doing, we cannot run your snippets as-is and we miss the entry points (the list of queues, coming from JSON files apparently).
